src/document.js

```diff
@@ -1167,15 +1173,12 @@ class DocumentTransform {
    * @returns {firestore.DocumentTransform}} The Document Transform.
    */
   static fromUpdateMap(ref, data) {
-    let transforms = [];
+    let transforms = new Map();
 
     function encode_(val, path, allowTransforms) {
-      if (val === FieldValue.SERVER_TIMESTAMP_SENTINEL) {
+      if (val instanceof FieldTransform && val.includeInDocumentTransform) {
         if (allowTransforms) {
-          transforms.push({
-            fieldPath: path.formattedName,
-            setToServerValue: SERVER_TIMESTAMP,
-          });
+          transforms.set(path, val);
         } else {
           throw new Error(
             'Server timestamps are not supported as array values.'
```

Comment:
Maybe this error should change now too, since this code is no longer specific to Server Timestamps?

Reply:
Made this error less specific.